expr/shaper.go

```diff
@@ -246,22 +246,30 @@ func NewShaperType(zctx *resolver.Context, fieldExpr Evaluator, typ *zng.TypeRec
 // to the provided typExpr. (typExpr should evaluate to a type value,
 // e.g. a value of type TypeType).
 func NewShaper(zctx *resolver.Context, fieldExpr, typExpr Evaluator, tf ShaperTransform) (*Shaper, error) {
-	lit, ok := typExpr.(*Literal)
-	if !ok {
-		return nil, fmt.Errorf("shaping functions (crop, fill, cast, order) take a literal as second parameter")
+	switch typExpr.(type) {
+	case *Var, *TypeFunc, *Literal:
+	default:
+		return nil, fmt.Errorf("shaping functions (crop, fill, cast, order) require a type value as second parameter")
```

Comment:
Nit: This function has some error strings that begin with "shaping functions (crop, fill, cast, order)" and others that begin with "shaper". Either's fine but consistency would be good.

Reply:
done